src/server/lock_manager/waiter_manager.rs

```diff
@@ -101,14 +101,12 @@ impl WaitTable {
 
     fn remove(&mut self, ts: u64) {
         self.wait_table.remove(&ts);
-        if self.wait_table.is_empty() {
-            self.has_waiter.store(false, Ordering::Relaxed);
-        }
+        self.waiter_count.fetch_sub(1, Ordering::SeqCst);
```

Comment:
You should decrease `waiter_count` in `get_ready_waiter()` and `remove_waiter()`.

Reply:
Sorry I misunderstood the original code, will fix.